### PR TITLE
[Issue 7864][docs]Update document for how to pick active consumer for failover subscription.

### LINKED ISSUE #7864: The master consumer is not chosen correctly in failover mode
**Describe the bug**
 In failover mode, the broker selects the master consumer based on connection order, not priority level and not the lexicographical sorting of a consumer name.

**To Reproduce**
1. Subscribe consumer1 with name "bbb" and PriorityLevel 1
2. Subscribe consumer2 with name "aaa" and PriorityLevel 0
3. See subscribe log:
[21:56:44.483 INF 9] consumer(1, bbb, -1) starting subscribe to topic persistent://public/default/topic-a
[21:56:44.519 INF 5] consumer(1, bbb, -1) subscribed to topic persistent://public/default/topic-a
[21:56:44.531 INF 5] consumer(2, aaa, -1) starting subscribe to topic persistent://public/default/topic-a
[21:56:44.570 INF 7] consumer(2, aaa, -1) subscribed to topic persistent://public/default/topic-a
[21:56:44.578 INF 7] consumer(2, aaa, -1) ActiveConsumerChanged isActive=False
[21:56:45.518 INF 6] consumer(2, aaa, -1) ActiveConsumerChanged isActive=False
[21:56:45.517 INF 7] consumer(1, bbb, -1) ActiveConsumerChanged isActive=True

**Expected behavior**
Active consumer must be consumer1 with name "aaa", but set consumer2 with name "bbb"
The same behavior is observed if the PriorityLevel of both consumer is set 0

**Desktop (please complete the following information):**
 - Broker run in docker apachepulsar/pulsar:2.6.0
 - Client: pulsar-client-dotnet on windows

**Additional**
Work correctly on docker apachepulsar/pulsar-test-latest-version:2.4.0
[22:52:02.086 INF 9] consumer(1, bbb, -1) starting subscribe to topic persistent://public/default/topic-a
[22:52:02.137 INF 6] consumer(1, bbb, -1) subscribed to topic persistent://public/default/topic-a
[22:52:02.149 INF 9] consumer(2, aaa, -1) starting subscribe to topic persistent://public/default/topic-a
[22:52:02.156 INF 5] consumer(2, aaa, -1) subscribed to topic persistent://public/default/topic-a
[22:52:03.140 INF 4] consumer(2, aaa, -1) ActiveConsumerChanged isActive=True
[22:52:03.140 INF 9] consumer(1, bbb, -1) ActiveConsumerChanged isActive=False

### PR DESCRIPTION
**Pull request by MarvinCai**
Update document to specify how active consumer is picked for partitioned and non-partitoned topic for failover subscription mode. The change was push in 2.4.2 but document has not been updated.

Fixes #7864

### CODE REVIEW

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -254,7 +254,7 @@ In the diagram below, only **Consumer A-0** is allowed to consume messages.
...
+In *failover* mode, multiple consumers can attach to the same subscription. For partitioned topic, in failover mode the broker selects the master consumer based on the priority level and the lexicographical sorting of a consumer name. If two consumers have an identical priority level, the broker selects the master consumer based on the lexicographical sorting. If these two consumers have different priority levels, the broker selects the consumer with a higher priority level as the master consumer. The master consumer is initially the only one receiving messages. When the master consumer disconnects, all (non-acknowledged and subsequent) messages are delivered to the next consumer in line. For non-partitioned topic in failover mode the broker will pick consumer in the order they subscribe to the non partitioned topic.
 
 For partitioned topics, the broker assigns partitioned topics to the consumer with the highest priority level. If multiple consumers have the highest priority level, the broker evenly assigns topics to consumers with these consumers.
```
Comment (Jennifer88huang) [inline site2/docs/concepts-messaging.md:260]: Suggestions from language perspectives:

In *failover* mode, multiple consumers can attach to the same subscription. In failover mode, the broker selects the master consumer for the partitioned topic based on the priority level and the lexicographical sorting of a consumer name. If two consumers have an identical priority level, the broker selects the master consumer based on the lexicographical sorting. If these two consumers have different priority levels, the broker selects the consumer with a higher priority level as the master consumer. Only the master consumer receives messages. When the master consumer is not connected, all (non-acknowledged and subsequent) messages are delivered to the next consumer in line. 

The broker assigns partitioned topics to the consumer with the highest priority level. If multiple consumers have the same priority level, the broker assigns topics to consumers evenly. 

In failover mode, the broker selects the master consumer for a non-partitioned topic based on the order that they subscribe to the non-partitioned topic.

Comment (Jennifer88huang) [reply site2/docs/concepts-messaging.md:260]: Do the following two have some conflicts?
1. If multiple consumers have the same priority level, the broker assigns topics to consumers evenly."
2.  If two consumers have an identical priority level, the broker selects the master consumer based on the lexicographical sorting.

Comment (MarvinCai) [reply site2/docs/concepts-messaging.md]: Thanks for the comment, yes it's conflicting description, I've updated the doc to reflect the latest way broker pick active consumer for topic in failover sub mode.

**Comment by Jennifer88huang** [comment]
@codelipenghui please help check whether the logic is correct, thank you.

**Comment by Jennifer88huang** [comment]
@MarvinCai Thanks for your contribution. You can use present tense rather than future tense, try to avoid using "will" where possible.
For details, refer to https://developers.google.com/style/tense

**Review by wolfstudy — APPROVED** [review]
Cool @MarvinCai , LGTM +1

File: site2/docs/concepts-messaging.md [hunk 2]
```diff
@@ -254,9 +254,11 @@ In the diagram below, only **Consumer A-0** is allowed to consume messages.
 
 ### Failover
 
-In *failover* mode, multiple consumers can attach to the same subscription. In failover mode, the broker selects the master consumer based on the priority level and the lexicographical sorting of a consumer name. If two consumers have an identical priority level, the broker selects the master consumer based on the lexicographical sorting. If these two consumers have different priority levels, the broker selects the consumer with a higher priority level as the master consumer. The master consumer is initially the only one receiving messages. When the master consumer disconnects, all (non-acknowledged and subsequent) messages are delivered to the next consumer in line.
+In *failover* mode, multiple consumers can attach to the same subscription. A master consumer will be picked for non-partitioned topic or each partition of partitioned topic. The master consumer is the only one receiving messages. When the master consumer disconnects, all (non-acknowledged and subsequent) messages are delivered to the next consumer in line.
```
Comment (Jennifer88huang) [inline site2/docs/concepts-messaging.md:257]: In *failover* mode, multiple consumers can attach to the same subscription. A master consumer is picked for topics and receives messages. When the master consumer disconnects, all (non-acknowledged and subsequent) messages are delivered to the next consumer in line.

Comment (MarvinCai) [reply site2/docs/concepts-messaging.md:257]: I'll keep the non-partitioned topic or each partition of partitioned topic part cuz I think reader will need this detail here.

Comment (MarvinCai) [reply site2/docs/concepts-messaging.md]: updated

**Review by gaoran10 — COMMENTED** [review]
LGTM +1

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**